uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/form/components/leave-type-wizard-form.component.js

```diff
@@ -12,7 +14,165 @@ define([], function () {
     $log.debug('Controller: LeaveTypeWizardFormController');
+
+    var vm = this;
+    var lineBreak = '\n\n';
+
+    vm.leaveTypeCategory = '';
+    vm.leaveTypeCategories = [
+      {
+        value: 'leave',
+        label: 'Leave',
+        icon: 'plane'
+      }
+    ];
+    vm.leaveTypeTitle = '';
+    vm.activeSettingsTabIndex = null;
+    vm.sections = [
+      {
+        name: 'general',
+        label: 'Leave Category'
+      },
+      {
+        name: 'settings',
+        label: 'Leave Category Settings'
+      }
+    ];
+    vm.sectionsTemplatesPath =
+      sharedSettings.sourcePath + 'leave-type-wizard/form/components/form-sections';
+    vm.settingsTabs = [
+      {
+        name: 'basic-details',
+        label: 'Basic',
+        fields: [
+          {
+            name: 'hide_label',
+            label: 'Hide leave type label on public calendars and feeds?',
+            helpText: [
+              'The CiviHR self service portal has an all staff calendar.',
+              'You can also create calendar feeds that can be integrated with your calendar app.',
+              lineBreak,
+              'If you enable this option the leave type will simply be given the title "Leave".',
+              'Managers and administrators will continue to see the actual leave title of the request.',
+              lineBreak,
+              'This may be helpful if the leave type title is sensitive - i.e. Unpaid leave or Compassionate leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'leave-requests',
+        label: 'Leave Requests',
+        fields: [
+          {
+            name: 'max_consecutive_leave_days',
+            label: 'Max consecutive duration (Leave blank for unlimited)',
+            helpText: [
+              'Configure the maximum duration of consecutive leave permitted to be selected in a single leave request.',
+              'You can leave this field blank for unlimited duration of leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'public-holidays',
+        label: 'Public Holidays',
+        fields: [
+          {
+            name: 'must_take_public_holiday_as_leave',
+            label: 'Do staff work on public holidays?'
+          }
+        ]
+      },
+      {
+        name: 'carry-forwards',
+        label: 'Carry Forwards',
+        fields: [
+          {
+            name: 'allow_carry_forward',
+            label: 'Allow carry forward?'
+          }
+        ]
+      }
+    ];
+
+    vm.$onInit = $onInit;
+    vm.getSettingsTabFields = getSettingsTabFields;
+    vm.openSection = openSection;
+    vm.openSettingsTab = openSettingsTab;
+    vm.selectLeaveTypeCategory = selectLeaveTypeCategory;
+
+    function $onInit () {
+      initDefaultView();
+    }
+
+    /**
+     * Gets fields for the currently active settings tab
+     *
+     * @return {Array} collection of fields
+     */
+    function getSettingsTabFields () {
```

Comment:
given my previous comment, this would be benefit from being made more generic, something like
```js
function getActiveTabFieldsOf(sectionName) {

}

getActiveTabFieldsOf('settings');
```
in any case the name is doubly misleading:
* it doesn't say anything about looking for the fields of the _active_ tab
* settings is a _section_ not a tab

Reply:
Renamed to `getFieldsForActiveSettingsTab`. It searches for fields for the active **tab** in Settings section